### ISSUE TITLE
🐛[BUG]: CorrDiff loss is scaled by hyper-parameter

### ISSUE THREAD

**Issue by chychen** [NVIDIA/modulus#605]
### Version

latest

### On which installation method(s) does this occur?

Source

### Describe the issue

CorrDiff loss is scaled by hyper-parameter, therefore we could not make a hyper-parameter search, because each run cannot be compared to the others.

example:
- if `batch_gpu_total` = 1, `loss_accum` = L, when `batch_gpu_total` = 2, `loss_accum` = L/2
- if `batch_size_gpu` = 1, `loss_accum` = L, when `batch_size_gpu` = 2, `loss_accum` = 2*L

why not just normalize it by `batch_size_global`? such as below
### Now Implementation
```python
      for round_idx in range(num_accumulation_rounds):
            with ddp_sync(ddp, (round_idx == num_accumulation_rounds - 1)):
                ...
                loss = loss.sum().mul(loss_scaling / batch_gpu_total)
                loss_accum += loss / num_accumulation_rounds
                loss.backward()

        loss_sum = torch.tensor([loss_accum], device=device)
        if dist.world_size > 1:
            torch.distributed.all_reduce(loss_sum, op=torch.distributed.ReduceOp.SUM)
        average_loss = loss_sum / dist.world_size
        if dist.rank == 0:
            wb.log({"training loss": average_loss}, step=cur_nimg)
```
### Proposed Modification
```python
      for round_idx in range(num_accumulation_rounds):
            with ddp_sync(ddp, (round_idx == num_accumulation_rounds - 1)):
                ...
                loss = loss.sum().mul(loss_scaling / batch_size_global) ### Modified
                loss_accum += loss ### Modified
                loss.backward()

        loss_sum = torch.tensor([loss_accum], device=device)
        if dist.world_size > 1:
            torch.distributed.all_reduce(loss_sum, op=torch.distributed.ReduceOp.SUM)
        average_loss = loss_sum / dist.world_size
        if dist.rank == 0:
            wb.log({"training loss": average_loss}, step=cur_nimg)
```

### Minimum reproducible example

```shell
see README
```

### Relevant log output

```shell
example:
- if `batch_gpu_total` = 1, `loss_accum` = L, when `batch_gpu_total` = 2, `loss_accum` = L/2
- if `batch_size_gpu` = 1, `loss_accum` = L, when `batch_size_gpu` = 2, `loss_accum` = 2*L
```

**Comment by mnabian** [comment]
Hi @chychen , thanks for reporting the issue. I agree with the proposed modification. Could you please open a PR?

**Comment by mnabian** [comment]
@chychen did you have a chance to make a PR for this modification?

**Comment by chychen** [comment]
this is an 3-month old issue, seems like the latest version has already solve this issue.

**Comment by chychen** [comment]
though the scale of losses are same now with different hyper-params, the gradients are still incorrectly scaled by "batch_size_per_gpu".

[now]
```python
            loss = loss.sum() / batch_size_per_gpu
            loss_accum += loss / num_accumulation_rounds
            loss.backward()

        loss_sum = torch.tensor([loss_accum], device=dist.device)
        if dist.world_size > 1:
            torch.distributed.barrier()
            torch.distributed.all_reduce(loss_sum, op=torch.distributed.ReduceOp.SUM)
        average_loss = (loss_sum / dist.world_size).cpu().item()

```

[proposed]
```python
            loss = loss.sum() / batch_size_per_gpu / num_accumulation_rounds/ dist.world_size #### modified, normalized by global batch size ####
            loss_accum += loss
            loss.backward()

        loss_sum = torch.tensor([loss_accum], device=dist.device)
        if dist.world_size > 1:
            torch.distributed.barrier()
            torch.distributed.all_reduce(loss_sum, op=torch.distributed.ReduceOp.SUM)
        average_loss = loss_sum.cpu().item() #### modified, normalized by global batch size ####
```
also do the same modification to the validation part.

does these modifications make sense?
